src/hooks.ts

```diff
@@ -311,3 +321,79 @@ export const useFeature: UseFeature = (featureKey, options = {}, overrides = {})
 
   return [state.isEnabled, state.variables, state.clientReady, state.didTimeout];
 };
+
+/**
+ * A React Hook that retrieves the flag decision, optionally
+ * auto updating those values based on underlying user or datafile changes.
+ *
+ * Note: The react client can become ready AFTER the timeout period.
+ *       ClientReady and DidTimeout provide signals to handle this scenario.
+ */
+export const useDecide: UseDecide = (flagKey, options = {}, overrides = {}) => {
+  const { optimizely, isServerSide, timeout } = useContext(OptimizelyContext);
+  if (!optimizely) {
+    throw new Error('optimizely prop must be supplied via a parent <OptimizelyProvider>');
+  }
+
+  const overrideAttrs = useCompareAttrsMemoize(overrides.overrideAttributes);
+  const getCurrentDecision: () => { decision: OptimizelyDecision } = useCallback(
+    () => ({
+      decision: optimizely.decide(flagKey, options.decideOptions, overrides.overrideUserId, overrideAttrs)
+    }),
+    [optimizely, flagKey, overrides.overrideUserId, overrideAttrs]
```

Comment:
Don't we need dependency on "options.decideOptions"?
If "IGNORE_USER_PROFILE" is added to the option, it'll generate different decision.

Reply:
yes, it makes sense. i fixed it.